coalahtml/_coalahtml/app/controllers/files.js

```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
     };
     $scope.fileName = $routeParams.fileName || $rootScope.ROOTHOME;
     parseCoalaProject($scope.fileName);
-    $scope.basename = function(path_name){
+
+    $scope.basename = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
       return path_name.split('/').slice(-1)[0];
     };
+    var len = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
+      return path_name.split('/').length;
+    };
+    $scope.getNameLink = function(fileName) {
+      var linkNameContainer = [], files = fileName.split('/');
+      var base = files.slice(0, len($rootScope.ROOTHOME)).join('/');
+      var prev = {
+        'name' : '/',
+        'link' : "#/file/" + base
+      };
+      linkNameContainer.push(prev);
+      var fileNameLen = len(fileName), rootHomeLen = len($rootScope.ROOTHOME);
+      for (var index = rootHomeLen; index < fileNameLen; index++) {
+        linkNameContainer.push({
+          'name' : index < fileNameLen-1 ? files[index] + prev.name : files[index],
+          'link' : prev.link + '/' + files[index]
+        });
+        prev.link = prev.link + '/' + files[index];
```

Comment:
You can modify `prev.link` before push and then push it. Then there will be no two: `prev.link + '/' + files[index]` pieces.